[Intel MKL] Adding support for MKL to docker CI infrastructure.

claynerobison opened the pull request:
This PR adds support for the MKL to the docker CI infrastructure. Non-development containers are supported via the publicly available pip whls at https://anaconda.org/intel/tensorflow/. Development containers are supported as well. 

This PR also adds support for the following parameters:

- CI_DOCKER_EXTRA_BUILD_PARAMS: allows callers to pass parameters to the docker build command (e.g. --build-arg)
- TF_BAZEL_BUILD_OPTIONS: allows callers to pass parameters to the bazel build command (e.g. "--copt='mavx2'")

The MKL Dockerfiles have been renamed per the convention in parameterized_docker_build.sh. They have also been modified to work better with the sed commands in the build scripts.

claynerobison committed:
Adding support for MKL to docker CI infrastructure.

inc0 commented on tensorflow/tools/docker/Dockerfile.mkl:
```diff
@@ -0,0 +1,70 @@
+FROM ubuntu:16.04
+
+LABEL maintainer="Craig Citro <craigcitro@google.com>"
```
aren't you maintainer Clayne?

claynerobison replied:
oh yeah.

inc0 commented on tensorflow/tools/docker/parameterized_docker_build.sh:
```diff
@@ -87,6 +87,10 @@
 #     (Optional)
 #     Specifies the desired build options. Defaults to OPT.
+#
+#   TF_BAZEL_BUILD_OPTIONS
```
you've introduced this env (and mechanism for it to work), but I can't see any place where you actually use it? Why?

claynerobison replied:
See line 332.

inc0 replied:
yeah you allowed to set TF_BAZEL_BUILD_OPTIONS and provided mechanism for it to work. I assume what will actually use this mechanism is part of CI somewhere? Not in this change?

claynerobison replied:
Correct. We have scripts internally that use it. tensorflow/tools/ci_build/linux/cpu/run_mkl.sh could (and probably should) use it. Right now, that script just takes default compiler x86 optimizations. tensorflow/tools/ci_build/ci_parameterized_build.sh has a TF_BUILD_OPTIONS parameter that is close, but doesn't give our scripts the flexibility they need.

inc0 replied:
OK, makes sense. Normally I'd say we should do it in separate PR as this is different feature/problem, but that's ok. Smaller PRs are just easier to review

claynerobison committed:
Update Dockerfile.mkl

rmlarsen commented:
@jhseu could you take a look, please?

jhseu commented:
The changes look good to me, but I'm switching reviewers to @gunan since there are test infrastructure changes going on, so I don't know if this is impacted.

gunan commented:
All of the bash files under here are on their way out. @case540 is working on removing those.
If you like to check in CI scripts, I recommend creating standalone scripts that do not need any flags under `tools/ci_build/linux/mkl`

claynerobison commented:
@case540 Can you give me some idea of what you are woring on  so I can help add MKL support to it?

case540 commented:
Yeah, working on rewriting these bash scripts in python to make the code more organized/readable.  Anything you add to the current .sh scripts will probably be deleted (or reimplemented in Python). 

Not much you can currently do to help except by changing the current build shell scripts as little as possible.

inc0 commented:
@case540 how about we merge this one (since python are still in the making) and we'll rewrite shells to .py as soon as you're ready?

case540 commented:
That would be fine with me.

yongtang committed:
Fix the issue with Bahdanau attention when normalized=True and dtype = float16/32

While revisiting 18016 I noticed that Bahdanau attention has a similiar
dtype mismatch issue when normalized=True. The issue comes from:
```
     g = variable_scope.get_variable(
         "attention_g", dtype=dtype,
         initializer=math.sqrt((1. / num_units)))
```
where the initializer value does not work well with differnt dtype.

This fix converts changes the initializer to `init_ops.constant_initializer`
to address the issue, and adds additional test cases for it.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for Bahdanau attention when normalized=True and dtype = float16/32

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix pylint issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

rmlarsen commented:
@case540 could you please review and approve this, if you are OK with merging it?

case540 commented on tensorflow/tools/docker/parameterized_docker_build.sh:
```diff
@@ -299,6 +317,26 @@ else
+
+  # Modify the devel Dockerfile with bazel build options
+  if [[ ! -z ${TF_BAZEL_BUILD_OPTIONS} ]]; then
+    if sed -i -e "s/--cxxopt=\"-D_GLIBCXX_USE_CXX11_ABI=0\" /--cxxopt=\"-D_GLIBCXX_USE_CXX11_ABI=0\" ${TF_BAZEL_BUILD_OPTIONS} /g" "${DOCKERFILE}"
```
Not really familiar with Dockerfiles, but is there a cleaner/less brittle way of modifying the Bazel build options?

claynerobison replied:
Same comment as above. I'm just following the pattern that the original authors were using, regardless of elegance. 

Dockerfiles themselves can have variables that can be overridden on the docker build command. For example, if the following line is in a Dockerfile:

`# ARGs are parameterized. Use --build-args to override.
+ARG WHL_DIR=/whl`

then when building the container, the caller can override the default value of WHL_DIR like this:

`docker build --build-arg WHL_DIR=<some path> . -f <docker file>`

That would be more elegant in my opinion.

case540 commented on tensorflow/tools/docker/parameterized_docker_build.sh:
```diff
@@ -299,6 +317,26 @@ else
 
+  # Modify the devel Dockerfile to specify the MKL bazel build parameters
+  if [[ "${TF_DOCKER_BUILD_TYPE}" == "mkl" ]]; then
+    if sed -i -e 's/bazel build -c opt /bazel build -c opt --config=mkl /g' "${DOCKERFILE}"
```
Not a huge fan of using sed on the dockerfile like this. Is there a better way to add in mkl build params?

claynerobison replied:
I don't think it's terribly elegant either, but it is what the original authors did for gpu builds in the lines just before this, so I thought it best to follow the model.

case540 replied:
I was hoping there was a cleaner way to write the dockerfile before merging this. Using "sed" on the very specific line in the docker file feels pretty brittle.

But, Im more than fine merging this before the python impl is ready.

yongtang committed:
Replace raw_input/input with six.moves.input

This fix is an enhancement to replace raw_input/input
in python 2 and 3 with six.moves.input.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Using xrange from six

In python 2 vs 3 xrange is different. This fix is an enhancement
to use xrange from six, instead of additional logic of handling
xrange in python 2 vs python 3.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 192856167

tensorflower-gardener committed:
Internal Change

PiperOrigin-RevId: 192856330

qlzh727 committed:
Merge pull request #18511 from qlzh727/branch_192842670

Branch 192842670

tensorflower-gardener committed:
Adding 1d and 3d orthogonal kernels convolutions.

PiperOrigin-RevId: 192861809

tensorflower-gardener committed:
Check there are no duplicate entries in sparse features as this would invalidate the example norm computation in SDCA.

PiperOrigin-RevId: 192861834

tensorflower-gardener committed:
Upgrade gRPC version used in OSS Tensorflow

PiperOrigin-RevId: 192862541

tensorflower-gardener committed:
[XLA] Use pattern matcher in algebraic simplifier

PiperOrigin-RevId: 192862841

facaiy committed:
Merge remote-tracking branch 'upstream/master' into ENH/adamax_optimizer

facaiy committed:
BLD: upgrade with eager.contex API

facaiy committed:
CLN: hide ApplyAdaMax op

tensorflower-gardener committed:
Make sure that same nodes are not optimized as part of multiple groups.

Replace recusrsion with iteration in AbsorbInputByOptimizedNodesGroup.

PiperOrigin-RevId: 192874364

tensorflower-gardener committed:
Style nit: avoid creating local variables when not required.

PiperOrigin-RevId: 192876802

yongtang committed:
Add shape validation for tag input of StatsDataset (#18462)

* Add shape validation for tag input of StatsDataset

The tag field of the StatsDataset needs to be a scalar.
However, there was no check in the shape function. This
fix adds the check of the tag shape.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add test case for tag shape check with BytesProducedStatsDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add tag shape check for LatencyStatsDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add test case for tag shape check with LatencyStatsDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Fix pylint issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update libpng to v1.6.34 for cmake

The libpng has been updated from v1.2.53 to v1.6.34 in PR 18299.
However, the cmake version of libpng has not been updated yet.
This fix updates the libpng for cmake to v1.6.34.

The fix is tested with cmake on linux:
```
tensorflow/tools/ci_build/ci_build.sh CMAKE tensorflow/tools/ci_build/builds/cmake.sh
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update library file names `libpng12` -> `libpng16`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add missing header pnglibconf.h

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

fumihwh committed:
move eye to linalg_ops_impl

fumihwh committed:
revert unneeded change

yongtang committed:
Update docs for tf.cast with supported types

This fix tries to address the issue raised in 18529 where
there are some confusion over supported types for `tf.cast`.
This fix updates the docs with explicitly supported numeric
type for `tf.cast`:
```
`uint8`, `int8`, `uint16`, `int16`, `int32`, `int64`,
`float16`, `float32`, `float64`, `complex64`, `complex128`, `bfloat16`.
```

This fix fixes 18529.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update sqlite version for cmake build (#18524)

The sqlite has been updated in bazel, though
cmake version was not updated. This fix updates
sqlite in cmake so that cmake and bazel versions
are synced.

The fix has been tested on Linux:
```
tensorflow/tools/ci_build/ci_build.sh CMAKE tensorflow/tools/ci_build/builds/cmake.sh
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Minor cleanup.

PiperOrigin-RevId: 192971080

ManHyuk committed:
Fix typo (#18416)

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* Improve comment

elsonrodriguez committed:
Improving S3 documentation. (#18406)

* Improving S3 documentation.

Added a copy-pastable guide on the variables, and also provided usable examples that give immediate feedback.

* Updating docs based on feedback.

Added back old configuration section, moved s3 implementations to bottom of document.

* Rearranged documentation before example, renamed sections to be more clear.

dthkao committed:
Expose Scaffold.default_local_init_op as a public static method. (#18398)

* Expose Scaffold.default_local_init_op as a public static method.

* update api

* Add a docstring.

* Add a returns section.

guillaumekln committed:
Support passing layer instances to produce attentional hidden states (#14974)

* Support passing Layer instances to the AttentionWrapper.

* Use _compute_output_shape to get the attention layer depth

* compute_output_shape is now a public method

* Move new argument at the end

yongtang committed:
Add support for explicit broadcasting in TensorFlow (#15243)

* Add support for explicit broadcasting in TensorFlow

This fix tries to adds support for explicit broadcasting in TensorFlow,
as was suggested in 14509. This fix adds the op of tf.broadcast_to,
which is equivalent to the numpy.broadcast_to in numpy.

This fix fixes 14509.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Register BroadcastTo op in array_ops.cc

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Sanitize with clang-format -i

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add test cases for tf.broadcast_to

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Sanitize bazel BUILD and python.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Split broadcast_to_ops_test from array_ops_test

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Support int64 shape

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Improve shape inference for broadcast_to

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add scalar input support for broadcast_to

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update API defs tensorflow/core/api_def/update_api_def.sh

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update API golden

```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update docstring for broadcast_to

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Enable GPU kernel for BroadcastTo

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Enable use_gpu=True for test cases

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Hiden the ops and export to tf.contrib.framework for now.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add the op to the _allowed_symbol in tf.contrib.framework

Otherwise the symbole will be hidden

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Fix pylint sanity issue.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

Commit:
adding ps_strategy to run_config to enable different placement strate… (#15640)

* adding ps_strategy to run_config to enable different placement strategy in estimator

* 1. Moved estimator._device_fn to RunConfig as @property
2. Made RunConfig.device_fn to return custom device function if one is specified, otherwise the result from `tf.train.replica_device_setter` call is used
3. Added some basic unit tests, may need further tests.

* 1. Removing ps_strategy.
2. Modified estimator to take overriden device_fn from  if set.
3. Removed ps_strategy related unit tests.

* Adding manual initialization of _device_fn in legacy RunConfig class

* Updated estimator golden API through
1. bazel build //tensorflow/tools/api/tests:api_compatibility_test
2. bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True

* fixing code styles

yongtang committed:
Allow `~/` in path for transform_graph (#15894)

* Allow `~/` in path for transform_graph

This fix tries to address the issue raised in 13211 where
it was not possible to specify `~` (e.g., `~/`, `~user/`, etc)
for the path used in transform_graph. This fix adds
the support of `~` transform_graph on Linux.

This fix fixes 13211.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Reformat transform_graph.cc with clang-format

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

freedomtan committed:
export tflite::Intepreter's  UseNNAPI() and setNumThreads() to java (#16065)

* export UseNNAPI() and setNumThreads() to java

Export tflite::Intepreter's UseNNAPI() and SetNumThreads() to Java
and modify the Android TfLiteCameraDemo app to use them.

* change CheckedChangeListener accordingly

* add error checking to setNumThreads()

terrytangyuan committed:
Fix embedding_ops doc formatting (#18520)

* Fix embedding_ops doc formatting

* Add missing indentation

tensorflower-gardener committed:
Internal change

PiperOrigin-RevId: 192981122

ManHyuk committed:
Merge branch 'fix_typo' of https://github.com/ManHyuk/tensorflow into fix_typo

ManHyuk committed:
fix typo

imsheridan committed:
Fix the doc strings of nn.sampled_softmax_loss since it was deprecated by nn.sampled_softmax_loss_v2

fumihwh committed:
remove unneeded sealing API code

jameswex committed:
Merge branch 'master' into closure_proto_library

jameswex committed:
Merge pull request #18500 from jameswex/closure_proto_library

closure proto library for example protos

imsheridan committed:
Fix tf.argmax warnings on dimension parameter with axis

Commit:
Update pydoc for several tfdbg hooks (#18533)

For classes:
Specify that tfdbg.LocalCLIDebugHook can be used to debug instances
of tf.estimator.Estimator. Remove mentions of
tf.contrib.learn.Estimator and tf.contrib.learn.Experiment, as
they are deprecated.

For __init__ method of LocalCLIDebugHook:
Clarify purpose of ui_type argument.

yongtang committed:
Update docs for return values as well

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update gemmlowp version for cmake build

The gemmlowp has been updated in bazel, though
cmake version was not updated. This fix updates
gemmlowp in cmake so that cmake and bazel versions
are synced.

The fix has been tested on Linux:
```
tensorflow/tools/ci_build/ci_build.sh CMAKE tensorflow/tools/ci_build/builds/cmake.sh
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

d0k committed:
Update for ObjectMemoryBuffer.h rename in upstream LLVM.

This will require a version bump in workspace.bzl

PiperOrigin-RevId: 193052084

Commit:
Add ability to override git tag in __git_version__ string.

Adding this functionality to make release process smoother. It will
allow us to create the release builds before creating the git
release tag.

yongtang committed:
Update docs for tf.cast with respect to complex types

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

annarev committed:
PiperOrigin-RevId: 193059174

Commit:
Fix Exception syntax.

annarev committed:
Merging r1.8 into master

annarev committed:
Prefix clip_by_value with underscore in gen_math_ops so that it doesn't
interfere with clip_ops.clip_by_value when importing.

PiperOrigin-RevId: 193069700

jsimsa committed:
Porting tests for the `decode_proto` and `encode_proto` to OS.

PiperOrigin-RevId: 193070420

isaprykin committed:
Remove the hidden replicate_model_fn copy from core.

PiperOrigin-RevId: 193070799

angerson committed:
Internal testing changes

PiperOrigin-RevId: 193071881

martinwicke committed:
Merge pull request #18544 from ManHyuk/fix_typo

Fix typo

tensorflower-gardener committed:
Refactoring: Rename the __ops module to ag__ (double underscore prefix has special meaning in Python). Consolidate all internal API calls through the ag__ module.

PiperOrigin-RevId: 193074379

tensorflower-gardener committed:
Mark the parent list as modified for index writes. Add special case for constructors where we know setting an attribute actually creates a new symbol. Clean up the tests a bit.

PiperOrigin-RevId: 193075909

martinwicke committed:
Merge pull request #18568 from case540/enable_git_tag_override

Add ability to override git tag in __git_version__ string.

tensorflower-gardener committed:
Remove obsolete tests. Patch the unexpected print output in Python 3.

PiperOrigin-RevId: 193078330

tensorflower-gardener committed:
Copy the if statement handlers over to the operators module. They will enabled in a follow-up CL.

PiperOrigin-RevId: 193078348

tensorflower-gardener committed:
Add a common transformer feature that allows keeping temporary state across nodes. To be used in the break, continue and return canonicalizers.

PiperOrigin-RevId: 193078359

tensorflower-gardener committed:
Tighten label check in BinaryLogisticHeadWithSigmoidCrossEntropyLoss

PiperOrigin-RevId: 193078844

martinwicke committed:
Merge pull request #17800 from yongtang/03152018-deprecated_argument_lookup

Enhancement with deprecated_argument_lookup for argmax/argmin

martinwicke committed:
Merge pull request #18569 from annarev/merge_branch

Merge branch

isaprykin committed:
Early TPU distribution strategy and the associated testing infrastructure.

PiperOrigin-RevId: 193080098

yongtang committed:
Update gemmlowp on bazel to sync cmake changes

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Temporary comment out mirror.bazel.build for gemmlowp

Will reenable once the mirror is propagated.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yk5 committed:
boosted_trees: Make some regularizer/hyper-params as inputs instead of attributes.

PiperOrigin-RevId: 193085059

rmlarsen commented:
@case540 & @claynerobison what is the status here? Can we proceed?
@claynerobison would you please rebase?

yongtang committed:
Add gemmlowp to whitelist

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Simplify the implementation of break_statements.py

PiperOrigin-RevId: 193086371

yongtang committed:
Update bazel

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 193089301

tensorflower-gardener committed:
Adding several utility functions to TF2XLA to help with the Cholesky refactor.  Mainly responsible for handling batching properly.

PiperOrigin-RevId: 193090634

nickdesaulniers committed:
[XLA] Document and enforce reduction order of init value

All existing backends apply the init_value on the lhs, except for the evaluator.  This causes problems for reductions which apply an identity function to a reduce or reduce window.

PiperOrigin-RevId: 193091323

martinwicke committed:
Merge pull request #18562 from yongtang/04152018-png

Update libpng to v1.6.34 for cmake build

tensorflower-gardener committed:
Simplify the recursion when processing unpackings.

PiperOrigin-RevId: 193094078

tensorflower-gardener committed:
Use fixed sized tensor arrays and max loop iterations in dynamic_decode if the user supplies it and if the inputs were created in an XLA context.

PiperOrigin-RevId: 193097293

itsmeolivia committed:
disabling test that fails tensorflow.asan

PiperOrigin-RevId: 193097794

martinwicke committed:
Merge pull request #18570 from yongtang/18106-Bahdanau-normalized-dtype

Fix the issue with Bahdanau attention when normalized=True and dtype = float16/32

martinwicke committed:
Merge pull request #18547 from imsheridan/fix_deprecated_softmax

Fix the doc strings of nn.sampled_softmax_loss since it was deprecated

martinwicke committed:
Merge pull request #18558 from imsheridan/fix_argmax_dimension

Fix tf.argmax warnings on dimension argument by using axis instead

yongtang committed:
Fix lite and makefile issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Exposes InputPipelineConfig as tf.contrib.tpu.InputPipelineConfig.
This type is expected by the `per_host_input_for_training` argument of the TPUConfig constructor, but is not currently visible.

PiperOrigin-RevId: 193101540

jsimsa committed:
Porting tests for `rpc_op` to OS.

PiperOrigin-RevId: 193102564

protoget committed:
Increase softmax gpu unittest numeric stability

PiperOrigin-RevId: 193103363

miaout17 committed:
Fix TFLite Makefile FFT2D dependency.

FFT2D dependency was introduced a while ago so Makefile no longer works
until this fix.

tensorflower-gardener committed:
Internal change.

PiperOrigin-RevId: 193112205

samsends committed:
Added support for saved_model_cli input files stored on GCS/AWS.

tensorflower-gardener committed:
Fix trace collection to properly remove the suffix.

PiperOrigin-RevId: 193113074

MarkDaoust committed:
Curly-brace id's are inconsistently supported.

Curly-brace id's are inconsistently supported.
linking to the id of an html tag seems to be supported everywhere.

MarkDaoust committed:
Merge branch 'master' into typo

tensorflower-gardener committed:
Replacing the current inner Cholesky decomposition loop with a While loop rolled version.

This will allow for much larger Cholesky decompositions (and thus matrix inversions) than previously possible on TPU because of the use of rolled While loops so XLA compilation will no longer timeout.

While there is a minor runtime performance decrease (now 25ms vs 15ms for a 500x500 matrix) the compilation time is significantly faster (12.8s vs 55.2s for a 500x500 matrix.)

PiperOrigin-RevId: 193114816

tensorflower-gardener committed:
Create copy of locals() before copying, since modifying locals does not always affect the values. https://docs.python.org/2/library/functions.html#locals.

PiperOrigin-RevId: 193116254

tensorflower-gardener committed:
Internal change

PiperOrigin-RevId: 193117142

itsmeolivia committed:
disabling flaky asan test

PiperOrigin-RevId: 193117611

martinwicke committed:
Merge pull request #17443 from yongtang/16263-softmax_cross_entropy-doc

Update the documentation of `softmax_cross_entropy`

tensorflower-gardener committed:
Port the list append into the operators module. Not enabled yet.

PiperOrigin-RevId: 193118940

annarev committed:
Internal change.

PiperOrigin-RevId: 193119953

martinwicke committed:
Merge pull request #18575 from miaout17/fix-fft

Fix TFLite Makefile FFT2D dependency.

tensorflower-gardener committed:
Add LinearOperatorKronecker, representing the Kronecker product.

PiperOrigin-RevId: 193123894

benoitsteiner committed:
Don't rely on graph contruction for an initial shape inference.

PiperOrigin-RevId: 193124836

mrry committed:
[tf.data] Sort the results of `tf.matching_files()` to enable `Dataset.list_files()` to be determinstic.

PiperOrigin-RevId: 193126572

Commit:
Remove deprecated/unused python related Bazel options.

Since py_runtime was introduced, Bazel ignores options such as
--force_python2 and --python2_path. Deleting to clean stuff up and
make sure people are not misled.

PiperOrigin-RevId: 193127681

samikama committed:
Fixes for review requests

yongtang committed:
Improve shape function check for `tf.roll`

The `tf.roll` op has requirements for the shape of inputs. However,
the shape of the inputs are only done at the runtime inside the kernel.
This fix improve the shape function so that the check could be
done early if shape is already known in the shape function.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for input shape of tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add axis shape check for tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for axis shape check with tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflower-gardener committed:
Adding min node weight regularization

PiperOrigin-RevId: 193131300

yongtang committed:
Add shape check for shift of tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for shape check with shift in tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

fumihwh committed:
Merge branch 'master' into support-matrix-2-norm

yongtang committed:
Check in shape function that axis and shift are same size

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for axis and shift shape equal check for tf.roll

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Sanitize with clang-format -i --style=Google

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint issue fix

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint issue fix

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Disable pylint: disable=redefined-builtin

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add shape check to TFRecordDataset

The inputs of TFRecordDataset have the requirements for shapes.
However, the check was not done in the shape function. This fix
adds shape checks whenever possible.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add shape check for compression_type in TFrecordDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

shashishekhar committed:
Add a simple Profiler and instrument operator invocations in Interpreter.

PiperOrigin-RevId: 193133955

yongtang committed:
Add shape check for buffer_size with TFRecordDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Sanitize with clan-format -i --style=Google

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yifeif committed:
Remove proto imports in header files for core/kernels/hexagon.

The goal is to make kernels mostly independent of proto headers, which will let us lock down our .so imports.

PiperOrigin-RevId: 193134710

ManHyuk committed:
fix typo

ManHyuk committed:
fix typo

ManHyuk committed:
fix typo

yk5 committed:
BoostedTreesEstimator in contrib: train_in_memory works with input_fns returning data.Dataset.
Only one batch of data is expected, so dataset.batch() is disallowed,
and dataset.repeat() will be ignored (only the first one would be used)

PiperOrigin-RevId: 193137094

ntenenz committed:
Remove conditional scope logic now that "current_arg_scope" exists in contrib

tensorflower-gardener committed:
Supply a dtype to super constructor, without which build() seems to crash.

PiperOrigin-RevId: 193139585

saxenasaurabh committed:
Make ParallelInterleaveDataset saveable.

PiperOrigin-RevId: 193142302

imsheridan committed:
Fix Warning in reduce_mean/ reduce_max related deprecation argument keep_dims

fumihwh committed:
fix init_ops.py

imsheridan committed:
revert unwanted typo

mrry committed:
[tf.data] Add an API for randomly sampling from multiple datasets.

Fixes #15999.

PiperOrigin-RevId: 193152683

fumihwh committed:
fix non-whitelited pylint errors

fumihwh committed:
loose test to 1e-5

fumihwh committed:
change the year of copyright

tensorflower-gardener committed:
Automated g4 rollback of changelist 192842670

PiperOrigin-RevId: 193168327

imsheridan committed:
Fix warnings for initialize_variables

tensorflower-gardener committed:
Fix the type info analysis to correctly process loops. Simplify the implementation by reusing some of the transformer base functions. Allow set_element_type to use literals. Add additional tests.

PiperOrigin-RevId: 193192409

tensorflower-gardener committed:
Add uint32/uint64 support to Gather op.

PiperOrigin-RevId: 193195939

dcirne committed:
Fix unintialized var warning in bfloat16

This contribution initializes result to 0, then inside the #if statement only one byte needs to be set, depending on the endian, the other will already be zero from the initialization. This also fixes the compilation warning.

tensorflower-gardener committed:
Fix incorrect rejection of xrange.

PiperOrigin-RevId: 193205016

yongtang committed:
Fix tf.compat.as_str returns bytes issue in Python 3

This fix tries to address the issue raised in 18598 where
tf.compat.as_str returns bytes (vs. str) in Python 3.
The issue was that `tf_export` decorator:
```
@tf_export('compat.as_bytes', 'compat.as_str')
```
could not be assigned to `as_bytes` or `as_text`
based on python 2 or 3.
This fix invokes tf_export explicitly based on `_six.PY2`
(for python 2/3) so that `as_str` calls `as_bytes` or `as_text`
conditionally.

This fix fixes 18598.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Removed unnneded tf_export

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint fix

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

Commit:
Estimate IdentityN as Identity.

PiperOrigin-RevId: 193207469

imsheridan committed:
Improve deprecation assignment with deprecated_argument_lookup

tensorflower-gardener committed:
Change the contract of dynamic_builtin to reject all functions it can't process.

PiperOrigin-RevId: 193215246

Commit:
Avoid ToString() in Eager's TFE_Execute.

Also use InlinedVector instead of std::vector for non-async path

Before:
Benchmark              Time(ns)        CPU(ns)     Iterations
-------------------------------------------------------------
BM_Execute/0               1895           1898         360200  Execute
BM_Execute/1               1193           1942         358322  ExecuteAsync
BM_ExecuteFunction/0       5812           5825         100000  ExecuteFunction
BM_ExecuteFunction/1       5015           5374         100000  ExecuteFunctionAsync

After:
Benchmark              Time(ns)        CPU(ns)     Iterations
-------------------------------------------------------------
BM_Execute/0               1604           1607         428262  Execute
BM_Execute/1               1150           1765         404821  ExecuteAsync
BM_ExecuteFunction/0       5615           5626         100000  ExecuteFunction
BM_ExecuteFunction/1       5111           5476         100000  ExecuteFunctionAsync
PiperOrigin-RevId: 193218331

yifeif committed:
Enable consumption of GIT_TAG_OVERRIDE env var in release build script. (#18579)

Enable consumption of GIT_TAG_OVERRIDE env var in release build script.

tensorflower-gardener committed:
Fixes a comment in tf.contrib.seq2seq.monotonic_attention().

PiperOrigin-RevId: 193224285

mingxingtan committed:
Replace decode_image with decode_jpeg to avoid ValueError in datasets programmers guide.

PiperOrigin-RevId: 193231717

martinwicke committed:
Merge pull request #18581 from ManHyuk/fix_typo

Fix typo

tensorflower-gardener committed:
Fixes and cleanup to support more complex quantized models and adds PropagateFakeQuantNumBits.

PiperOrigin-RevId: 193232630

martinwicke committed:
Merge pull request #18601 from yongtang/18598-tf.compat.as_str

Fix tf.compat.as_str returns bytes issue in Python 3

pavithrasv committed:
Moving gradient registration for CudnnRNN op from contrib to core.

PiperOrigin-RevId: 193234663

yifeif committed:
Automated g4 rollback of changelist 193168327

PiperOrigin-RevId: 193234819

LoSealL committed:
Complement cmake script to compile tensorflow with mkl and mkldnn on Windows (#16936)

* Add build batch for windows

* Automaticaly find python, cuda, mkl runtimes in PATH

* auto select cmake generator

* Add external library mkldnn. Add options for mkl and mkldnn

* fix syntax error in make.bat

* Fix errorlevel syntex bug in make.bat

* Add /arch:avx2 flags to enable avx2 on windows

* Revert to keep `tensprflow_WIN_CPU_SIMD_OPTIONS` unchanged, add an option `tensorflow_ENABLE_MKL_SUPPORT` to include MKL compilation. Still specify SIMD flags by setting `tensorflow_WIN_CPU_SIMD_OPTIONS` to such as '/arch:AVX2'

* Fix a mistake of CUDA path in make.bat

* resolve conflict in mkl_cpu_allocator.h

* Improve error detection

* Use where /Q to detect cmd environment

* fix "ELSE IF" the syntax error in make.bat

* update README.md, wrap windows based codes by #ifdef _WIN32

* unistd.h is not needed in mkl_cpu_allocator.h any more in master branch

* Remove inline of kMaxLimitStr, which causes compile error in VS2015

* Add static_cast in  to fix compile error

* remove make.bat

* Removed make.bat description part

yifeif committed:
Merge commit for internal changes

yongtang committed:
Remove duplicate import in compat.py

Noticed there are a couple of places in compat.py that
have duplicate import:
```
from tensorflow.python.util.tf_export import tf_export
from tensorflow.python.util.tf_export import tf_export
```

This fix remove duplicate imports.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Duplicate imports in histogram_ops.py

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

annarev committed:
Merge pull request #17395 from facaiy/ENH/adamax_optimizer

add AdaMax optimizer

yifeif committed:
Update clip_ops_test.py

martinwicke committed:
Merge pull request #17512 from imsheridan/fix_broken_link_graph

Fix broken link of distributed in graphs.md

martinwicke committed:
Merge pull request #17562 from giuscri/typo

Fix markdown error in layers tutorial.

martinwicke committed:
Remove range_builder

martinwicke committed:
Merge pull request #18563 from yongtang/04162018-gemmlowp

Update gemmlowp version for cmake build

martinwicke committed:
Merge pull request #18561 from yongtang/18529-doc-cast

Update docs for tf.cast with supported types

dcirne committed:
QueueOptions size var data types to size_t

QueueOptions' max_batch_size and max_enqueued_batches are positive quantities, and when compared, in the code, with unsigned member functions, a warning is raised. By changing the data type from int to size_t, not only the meaning of the member variables are more aligned with their intent, but also the comparisons are done between unsigned integers, thus fixing the warnings.

martinwicke committed:
Merge pull request #18614 from yongtang/04172018-duplicate-import

Remove duplicate import in compat.py

yongtang committed:
Remove duplicate imports in several places

Wrote a script to scan throught the python files in the repo,
and found the remaining duplicate imports in some python files like:
```
 from tensorflow.python.ops import random_ops
-from tensorflow.python.ops import random_ops
 from tensorflow.python.util.deprecation import deprecated
```
This fix removed all of them for duplicate imports.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix duplicate import in kmeans_test.py

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Clean up remaining issues.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

akshaym committed:
Merge pull request #18504 from akshaym/eager-custom-ops

Use eager compatible wrappers in load_library for custom ops

yifeif committed:
Merge pull request #18613 from yifeif/branch_193234819

Branch 193234819

martinwicke committed:
Merge pull request #18615 from yongtang/04162018-TFRecordDataset-shape

Add shape check to TFRecordDataset

mrry committed:
[tf.data] Fix a device placement issue in `prefetch_to_device()`. (#18607)

* [tf.data] Fix a device placement issue in `prefetch_to_device()`.

Previously, the `iterator_get_device()` op was being infeasibly colocated with
both the iterator and placed on the prefetch target device. Move the
construction of that op outside the `with device():` block to fix this.

Also enable the relevant test to run as a CUDA test.

* Import the cuda_py_test rule.

samikama committed:
Make requested review changes

martinwicke committed:
Merge pull request #18618 from yongtang/04172018-duplicate-random_ops

Remove duplicate imports in all python source

aaroey committed:
Fix py_test import

agramesh1 committed:
[INTEL MK] Updating MKL CPU CI build and test. (#18513)

* Setting  KMP_BLOCKTIME to 0

* Adding comments

shengfuintel committed:
Fixed the bug in mkl_input_conversion_op when reorder is not needed (#18498)

yongtang committed:
Fix shape validation error with tf.nn.conv3d_transpose (#18465)

* Fix shape validation error with tf.nn.conv3d_transpose

This fix tries to address the issue raised in 18460.
In `tf.nn.conv3d_transpose` when list or np array is passed,
the validate of the output shape with filter shape uses
`output_shape[4]` (channel). This will not work with
`data_format='NCDHW'`.

This fix fixes the issue by replace `output_shape[4]` with `output_shape[axis]`.

This fix also adds a test case. Before this fix, the test case will fail.

This fix fixes 18460.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add test case for output and filter shape check in conv3d_transpose

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Fix pylint issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Also fix the error message

yongtang committed:
Replace raw_input/input with six.moves.input for python 2/3 (#18461)

In python 3 input is the equivalent of raw_input in python 2.
This fix is an enhancement to replace raw_input/input with
six.moves.input, which is compatible with both python 2 and python 3.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

imsheridan committed:
Fix the default value doc string of global_step in contrib.slim (#18313)

fo40225 committed:
fix build break cmake windows 32bit (#18295)

* fix build break cmake windows 32bit

* Fix lint errors

yifeif committed:
Merge pull request #18577 from sbsends/master

Added support for saved_model_cli input files stored on GCS/AWS.

yongtang committed:
Add uint16 support for py_func

In tf most of the numeric data types are supported though uint16 support
is not:
```
$ python
>>> import tensorflow as tf
>>> def sum_func(x, y):
...   return x + y
...
>>> x = tf.constant(1, dtype=tf.uint16)
>>> y = tf.constant(2, dtype=tf.uint16)
>>> z = tf.py_func(sum_func, [x, y], tf.uint16)
>>> tf.Session().run(z)
...
...
tensorflow.python.framework.errors_impl.UnimplementedError: Unsupported numpy type 4
	 [[Node: PyFunc = PyFunc[Tin=[DT_UINT16, DT_UINT16], Tout=[DT_UINT16], token="pyfunc_0", _device="/job:localhost/replica:0/task:0/device:CPU:0"](Const, Const_1)]]
...
```

The reason is that there is no conversion between numpy uint16 and tf.uint16.

This fix adds the support so that py_func could process tf.uint16 data types.

This fix also adds test cases for different data types with py_func to
increase the test coverage.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test cases for real data types with py_func.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix line too long issue with pylint

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test cases for complex (complex64/complex128) type with py_func

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint fix

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

annarev committed:
Merge pull request #17740 from fumihwh/support-matrix-2-norm

implement matrix 2-norm

yongtang committed:
Update complex test case

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for py_func with bool types

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Remove unneeded for loop for bool data types in tests

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update dtypes for the test cases

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint fix for the import

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Remove unneeded assignment

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix `Line too long (82/80) (line-too-long)` issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

aaroey committed:
Merge pull request #18509 from samikama/py_test

Add testing for TF-TRT

langner committed:
Remove over-indentation

yesmung committed:
Fix wrong api name in apis.md

typed_output_tensor is the correct api name.
It is implemented in the interpreter class.

Signed-off-by: MyungSung Kwak <yesmung@gmail.com>

imsheridan committed:
Fix useless duplicate lines in *.py files

joeyearsley committed:
Update debugger.md

Should be using normal softmax not sparse.

imsheridan committed:
Add shape check to TextLineDataset op

martinwicke committed:
Merge pull request #17032 from yongtang/02152018-keep_dims-monte_carlo_test

Fix warnings in tf.contrib.bayesflow.monte_carlo.expectation

imsheridan committed:
add checks for compression_type and buffer_size also

yongtang committed:
Fix build failure in `bazel test -s --config=opt --cache_test_results=no //tensorflow/python/kernel_tests:init_ops_test`

With the most recent master the following test fails:
```
bazel test -s --config=opt --cache_test_results=no //tensorflow/python/kernel_tests:init_ops_test
...
...
...
    eye = linalg_ops.eye(n, dtype=self.dtype)
NameError: global name 'linalg_ops' is not defined
```

This fix fixes the test failure.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yifeif committed:
Merge pull request #18664 from yongtang/04182018-init_ops.py

Fix build failure in `//tensorflow/python/kernel_tests:init_ops_test`

yongtang committed:
Fix issue for float16 data type with reuse in CudnnLSTM

This fix tries to address the issue raised in 18699 where
for float16 data type, the reuse in CudnnLSTM throws a ValueError.

This fix fixes the issue by passing the data type. This fix
fixes 18699.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix pylint `Line too long (102/80)`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update comment in the test

`checked is done in kernel.` -> `checked in kernel.`

for review feedback.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

samikama committed:
Configurable Custom graph optimizers for grappler (#18479)

* Adding a new field for configurable custom optimizers

* Pass configuration structure to custom optimizers

* Reviewer requests

* Style fixes

* Fix tests

yongtang committed:
Add string_strip to remove leading and trailing whitespaces (#18418)

* Add string_strip to remove leading and trailing whitespaces

This fix tries to address the issue raised in 18384 to add
an op tf.string_strip so that the leading and trailing whitespaces
could be removed.

This fix fixes 18384.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add StringStrip op to string_ops.cc

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update Bazel BUILD file

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Sanitize with clang-format

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add test cases for string_strip

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Fix pylint issue

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update API defs with tensorflow/core/api_def/update_api_def.sh

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Fix python test error

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Update goldens API with

```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

martinwicke committed:
Merge pull request #18611 from yongtang/04162018-roll-shape

Improve shape function check for `tf.roll`

ManHyuk committed:
fix typo

martinwicke committed:
Merge pull request #18674 from yongtang/18669-CudnnLSTM

Fix issue for float16 data type with reuse in CudnnLSTM

martinwicke committed:
Merge pull request #18659 from yongtang/04182018-py_func_uint16

Add uint16 support for py_func

martinwicke committed:
Merge pull request #18657 from yongtang/04152018-raw_input

Replace raw_input/input with six.moves.input

martinwicke committed:
Merge pull request #18650 from imsheridan/add_shape_check_TextLineDataset

Add shape check to TextLineDataset op

yongtang committed:
Add shape check for batch related Dataset ops (#18683)

* Add shape check for PrefetchDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add BatchDataset shape check

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for SlideDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for DenseToSparseBatchDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Sanitize with clang-format -i --style=Google

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Shape validation with random/shuffle related Dataset ops (#18682)

* Add shape check for CacheDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for ShuffleAndRepeatDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add check for ShuffleDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for RandomDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add RangeDataset shape check

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Sanitize with clang-format -i --style=Google

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Validation in shape functions of Dataset ops (#18680)

* Add shape check for PrependFromQueueAndPaddedBatchDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add comment for shape check

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for FixedLengthRecordDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add check for filenames as well

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Clang-format -i --style=google for file format

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

* Add shape check for SqlDataset

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

martinwicke committed:
Merge pull request #18632 from yesmung/master

Fix wrong api name in apis.md

martinwicke committed:
Merge pull request #18610 from yongtang/04152018-xrange-six

Using xrange from six

martinwicke committed:
Merge pull request #17815 from brettkoonce/data_spelling

contrib: minor spelling tweaks

yifeif committed:
Fix doc gen error

Mismatch after the fix in #17815

imsheridan committed:
Fix minor pylint issue

martinwicke committed:
Merge pull request #18599 from dcirne/bfloat16

Fix unintialized var warning in bfloat16

martinwicke committed:
Merge pull request #18584 from imsheridan/fix_reduce_keepdims

Fix Warning in reduce_* related deprecation argument keep_dims

martinwicke committed:
Merge pull request #18603 from imsheridan/fix_variables_initializer

Fix deprecateion warnings for tf.initialize_variables

martinwicke committed:
Merge pull request #18639 from imsheridan/fix_duplicate_lines

Remove useless duplicate lines in *.py files

martinwicke committed:
Merge pull request #18655 from langner/patch-1

Remove over-indentation

Commit:
Merge pull request #18701 from tensorflow/yifeif-patch-3

Fix doc gen error

joel-shor committed:
[tf.data] Allow `sample_from_datasets` to accept a tf.Dataset object for `weights`.

Tested:
bazel test :interleave_dataset_op_test

joel-shor committed:
[tf.data] Correct / clarify docstring for `weights` as a dataset.

This is a noop.

martinwicke committed:
Merge pull request #18605 from imsheridan/improve_deprecated_argument

Improve deprecation assignment with deprecated_argument_lookup

martinwicke committed:
Merge pull request #18616 from dcirne/batch_size

QueueOptions var data types to size_t

martinwicke committed:
Merge pull request #18646 from joeyearsley/patch-4

Update debugger.md

martinwicke committed:
Merge pull request #18686 from ManHyuk/fix_typo

fix typo

martinwicke committed:
Merge pull request #18582 from ntenenz/master

Remove conditional scope logic now that "current_arg_scope" exists in contrib

joel-shor committed:
Merge pull request #18707 from joel-shor/master

[tf.data] Allow `sample_from_datasets` to accept a tf.Dataset object …

claynerobison committed:
Rebasing

claynerobison commented:
Closing because the rebase was a disaster. See new PR #18745
